GG-32298 Create a JMX metric to show realtime sql engine memory utililization

akorensh opened the pull request:
…zation

akorensh committed:
GG-32298 Create a JMX metric to show realtime sql engine memory utilization

slukyano reviewed (changes requested):
Please add a test case which checks that the metric value is as expected when queries are executing.

I’d do something like this:

1. Check that `freeMem == quota`
2. Execute some heavy SQL queries
3. During the query execution check that freeMem is as expected (it’s OK to check with low precision, like freeMem > 10MB && freeMem < 20MB
3. Check that `freeMem == quota` after the query has completed
4. Execute some SQL queries to provoke quota overflow
5. Check that `freeMem == quota` after the queries have been canceled due to quota overflow.

slukyano reviewed (commented):
I see that the javadocs for the quotas are incorrect as well. May I ask you to also fix them in this patch? At least change getSqlQueryMemotyQuota[Bytes]() javadoc to "per-query quota" from "global query quota"

akorensh committed:
GG-32298 Create a JMX metric to show realtime sql engine memory utilization

akorensh committed:
GG-32298 Create a JMX metric to show realtime sql engine memory utilization
change freeMem and maxMem to be sourced directly from QueryMemoryManager

akorensh committed:
GG-32298 Create a JMX metric to show realtime sql engine memory utilization
change freeMem and maxMem to be sourced directly from QueryMemoryManager

akorensh committed:
GG-32298 Create a JMX metric to show realtime sql engine memory utilization
change freeMem and maxMem to be sourced directly from QueryMemoryManager- style fixes

akorensh committed:
GG-32298 Create a JMX metric to show realtime sql engine memory utilization
change freeMem and maxMem to be sourced directly from QueryMemoryManager - style fixes
